include/swift/AST/ArchetypeBuilder.h

```diff
@@ -302,7 +302,9 @@ class ArchetypeBuilder {
   ///
   /// \param allowConcreteGenericParams If true, allow generic parameters to
   /// be made concrete.
-  void finalize(SourceLoc loc, bool allowConcreteGenericParams=false);
+  void finalize(SourceLoc loc,
+                ArrayRef<GenericTypeParamType *> genericParams,
```

Comment:
They're supposed to go away from inside the archetype builder. My plan here is that the archetype builder should be canonical, and we feed in the specific generic parameters only in those places where we might need to emit a diagnostic.